src/Workspaces/Core/MSBuild/MSBuild/BuildHostProcessManager.cs

```diff
@@ -134,10 +140,12 @@ private ProcessStartInfo CreateDotNetCoreBuildHostStartInfo()
         // We need to roll forward to the latest runtime, since the project may be using an SDK (or an SDK required runtime) newer than we ourselves built with.
         // We set the environment variable since --roll-forward LatestMajor doesn't roll forward to prerelease SDKs otherwise.
         processStartInfo.Environment["DOTNET_ROLL_FORWARD_TO_PRERELEASE"] = "1";
-        processStartInfo.ArgumentList.Add("--roll-forward");
-        processStartInfo.ArgumentList.Add("LatestMajor");
+        AddArgument(processStartInfo, "--roll-forward");
+        AddArgument(processStartInfo, "LatestMajor");
 
-        processStartInfo.ArgumentList.Add(typeof(IBuildHost).Assembly.Location);
+        var netCoreBuildHostPath = Path.Combine(Path.GetDirectoryName(typeof(BuildHostProcessManager).Assembly.Location)!, "BuildHost-net6.0", "Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost.dll");
```

Comment:
ok to hardcode these values in?  the `-net6.0` part seems fishy to me

Reply:
These match the names of the folders we deploy in the NuGet package, so yes, these can be hardcoded.

Reply:
consider doc'ing :)

Reply:
Will defer to follow-up PR.